overlays/bootstrap.nix

```diff
@@ -366,11 +366,23 @@ in {
 
                 src-spec = rec {
                     version = "8.10.2";
-                    url = "https://downloads.haskell.org/~ghc/${version}/ghc-${version}-src.tar.xz";
-                    sha256 = "02w8n085bw38vyp694j0lfk5wcnwkdaj7hhp0saj71x74533lmww";
+                    file = final.runCommand "ghc-${version}-src" {
+                      src = final.fetchgit {
+                        url = "https://gitlab.haskell.org/ghc/ghc.git";
+                        rev = "b0ad86fb84fbd2ac78208e6545c48c7a09e7f4aa";
+                        sha256 = "18sq4bn7wfdmpn04gqxbbcpdlkg7sm8qhylfwc0l6mpry74lybh2";
+                      };
+                    } ''
+                        cp -r --no-preserve=mode $src $out
+                        PATH=$PATH:${final.autoconf}/bin:${final.automake}/bin
+                        cd $out
+                        ${final.python3}/bin/python ./boot
+                      '';
                 };
 
                 ghc-patches = ghc-patches "8.10.2";
+
+                ghc-version = "8.10.2.20201212";
```

Comment:
We should also make it so the `compiler-nix-name` matches (instead of `ghc8102 = final.callPackage ../compilerghc {` have `ghc810220201212 = final.callPackage ../compilerghc {`).

There is code in `plan-to-nix` that takes the numeric version reported by ghc, removes the `.`s and sets `compiler.nix-name` with it.